src/mlpack/core/cv/metrics/ssim_index_impl.hpp

```diff
@@ -0,0 +1,89 @@
+/**
+ * @file ssim_index.hpp
+ * @author Utkarsh Rai
+ *
+ * The SSIM (Structural Similarity) metric.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_METRICS_SSIMINDEX_IMPL_HPP
+#define MLPACK_CORE_CV_METRICS_SSIMINDEX_IMPL_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace cv {
+
+template<typename DataType>
+double SSIMIndex::Evaluate(DataType const& image,
+                           DataType const& reference)
+{
+  if (arma::size(reference) != arma::size(image))
+  {
+    std::ostringstream oss;
+    oss << "SSIMIndex::Evaluate(): size of reference ("
+      << reference.n_cols << "X" << reference.n_rows
```

Comment:
This should be tabbed twice (4 spaces), see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#line-length-and-wrapping for more details.